lib/datadog/appsec/configuration/settings.rb

```diff
@@ -199,10 +206,10 @@ def merge(dsl)
 
             # TODO: move to a separate apply step
             klass = registered_integration.klass
-            if klass.loaded? && klass.compatible?
-              instance = klass.new
-              instance.patcher.patch
-            end
+            next unless klass.loaded? && klass.compatible?
+
+            instance = klass.new
+            instance.patcher.patch unless instance.patcher.patched?
```

Comment:
Curious about `unless instance.patcher.patched?`

IIRC `instance.patcher.patch` should be idempotent

Reply:
Makes sense